build-system/tasks/extension-generator/extension-doc.template.md

```diff
@@ -0,0 +1,220 @@
+ -->
+
+[filter formats=“ads”]
+
+Below is an example for ads.
+
+[example preview=“inline” playground=“true” imports="${name}"]
+
+```html
+<${name} required-attribute>
+  I am a hello world inline executable code sample for ads!
+</${name}>
+```
+
+[/example][/filter]
+
+### Standalone use outside valid AMP documents (optional)
+
+Bento AMP allows you to use AMP components in non-AMP pages without needing to commit to fully valid AMP. You can take these components and place them in implementations with frameworks and CMSs that don't support AMP. Read more in our guide [Use AMP components in non-AMP pages](https://amp.dev/documentation/guides-and-tutorials/start/bento_guide/)
+
+#### Example
+
+Code snippet of Bento component in standalone usage.
+
+#### Interactivity and API usage
+
+Bento enabled components in standalone use are highly interactive through their API. Because standalone mode doesn't require valid AMP, you can create custom interactivity without the use of [`amp-bind`](https://amp.dev/documentation/components/amp-bind/?format=websites) implementations and the API replaces [AMP actions and events](https://amp.dev/documentation/guides-and-tutorials/learn/amp-actions-and-events/?format=websites).
+
+The `${name}` component API is accessible by including the following script tag in your document:
+
+```
+await customElements.whenDefined('${name}-component');
+const api = await ${name}.getApi();
+```
+
+The `${name}` API allows you to register and respond to the following events:
+
+**event 1**
+Explanation of event, proper syntax/arguments.
+
+```
+example
+```
+
+**event 2**
+Explanation of event, proper syntax/arguments.
+
+```
+example
+```
+
+#### Layout and style
+
+Fully valid AMP pages use the AMP layout system to infer sizing of elements to create a page structure before downloading any remote resources. However, Bento use imports components into less controlled environments and AMP's layout system is inaccessible.
+
+**Container type**
+
+The `${name}` component has a container/non-container layout type. To ensure the component renders correctly, apply the following styles:
+
+```css
+example
+```
+
+**style/layout guidelines 2 (optional)**
+
+Information on how to layout and style `${name}`.
+
+```
+example
+```
+
+### Behavior users should be aware of (optional)
+
+What to do if they want behavior. How to work around it.
+
+```html
+<${name} required-attribute>
+  Code sample of behavior or behavior workaround.
+</${name}>
+```
+
+### Behavior restrictions
+
+What is allowed, what isn't.
+
+## Attributes
+
+### `attribute-name`
+
+Description of attribute. Use cases for this attribute.
+
+-   `attribute-value-option-one` (default): `attribute-option-one-value` does this to `${name}`.
+-   `attribute-value-option-two`: `attribute-option-two-value` does this to `${name}`.
+
+### `optional-attribute-name` (optional)
+
+Here, I write what `optional-attribute-name` will do to `${name}`.
```

Comment:
suggestion: could be good to have a specific format for how to phrase this here, i.e. "`optional-attribute-name` enables ____ when present. By default if omitted, ${} will behave as ____`"